core/include/detray/utils/grid/detail/axis.hpp

```diff
@@ -83,6 +83,16 @@ struct single_axis {
                                    const vector_type<scalar_type> *edges)
         : m_binning(indx_range, edges) {}
 
+    /// Equality operator
+    ///
+    /// @param rhs is the right-hand side of the comparison
+    ///
+    /// @returns whether the two axes are equal
+    DETRAY_HOST_DEVICE constexpr auto operator==(const single_axis &rhs) const
+        -> bool {
+        return m_bounds == rhs.m_bounds && m_binning == rhs.m_binning;
```

Comment:
The bounds type does not hold any runtime data, do we really need to compare it? I.e. I think the bounds can only ever be different if ```rhs``` was a different ```single_axis``` type